Lib/zipfile.py

```diff
@@ -2004,26 +2059,56 @@ def main(args=None):
     elif args.create is not None:
         zip_name = args.create.pop(0)
         files = args.create
-
+        mode = 'w'
+        if args.verbose:
+            print('Create {!r}:'.format(zip_name))
+
+    elif args.append is not None:
+        zip_name = args.append.pop(0)
+        files = args.append
+        mode = 'a'
+        if args.verbose:
+            print('Add to {!r}:'.format(zip_name))
+
+    if mode and zip_name:  # Create or append files
+        import glob
         def addToZip(zf, path, zippath):
+            count = 0
             if os.path.isfile(path):
+                if args.verbose:
+                    print("Add: {!r}".format(zippath))
                 zf.write(path, zippath, ZIP_DEFLATED)
+                count += 1
             elif os.path.isdir(path):
-                if zippath:
-                    zf.write(path, zippath)
-                for nm in os.listdir(path):
-                    addToZip(zf,
-                             os.path.join(path, nm), os.path.join(zippath, nm))
+                if not os.path.split(path)[-1].startswith('.'):
```

Comment:
glob.iglob doesn't have an option to give me this information - I was trying to avoid using os.walk and walking, filtering myself. Added:
```
                        if file_name.startswith('**/'):
                            print("Recursing directories that don't start with .")
                        elif file_name.startswith('.**/'):
                            print("Recursing directories that do start with .")
```